src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const stateCopy = {
+    ...state,
+  };
+
+  actions.forEach(object => {
+    switch (object.type) {
+      case 'addProperties':
+        for (const key in object.extraData) {
+          stateCopy[key] = object.extraData[key];
+        }
+        break;
+      case 'removeProperties':
+        object.keysToRemove.forEach(key => {
+          delete stateCopy[key];
+        });
+        break;
+
+      default:
```

Comment:
The same comment as in the previous task - https://github.com/mate-academy/js_stateful-object/pull/2187#discussion_r1138825291